numba_dpex/core/utils/kernel_launcher.py

```diff
@@ -430,10 +530,52 @@ def set_range(
             types.uintp, len(global_range)
         )
 
+    def set_range_from_indexer(
+        self,
+        ty_indexer_arg: Union[RangeType, NdRangeType],
+        ll_index_arg: llvmir.BaseStructType,
+    ):
+        """Returns two lists of LLVM IR Values that hold the unboxed extents of
+        a Python Range or NdRange object.
+        """
+        ndim = ty_indexer_arg.ndim
+        global_range_extents = []
+        local_range_extents = []
+        indexer_datamodel = self.context.data_model_manager.lookup(
+            ty_indexer_arg
+        )
+
+        if isinstance(ty_indexer_arg, RangeType):
+            for dim_num in range(ndim):
+                dim_pos = indexer_datamodel.get_field_position(
+                    "dim" + str(dim_num)
+                )
+                global_range_extents.append(
```

Comment:
Unrelated to the changes here, but the index order has to be reversed. I am mentioning here so that we keep it in mind.

Index inversion should be done here as numba-dpex first compiles a native kernel (OpenCL or Level Zero) and then generates a SYCL interoperability kernel from it. The convention for unit stride dimensions is opposite for OpenCL and SYCL (refer: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:opencl:kernel-conventions-sycl). For this reason, although numba-dpex follows SYCL like indexing in the user-facing front-end while launching the kernel the indexing has to be reversed. 

We do the same thing in `core.kernel_interface.dispatcher.JitKernel.__getitem__`

Reply:
Okay! Let's do it in separate PR. This PR does not mean to change logic.

Reply:
Yes, sure. I created a ticket to keep track.